### PR TITLE
TIP-1234: enhance documentation about documentation ^^

### PR DESCRIPTION
**Pull request by jjanvier**
This PR allows to deploy master documentation.
The need for a fully installed PIM with MySQL has been dropped.
It's still not perfect, a lot of other things could be removed, but it's better than it was.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -39,13 +30,11 @@ RUN apt-get update && \
     #
     # Download packages
     apt-get update && \
-    apt-get install -y --no-install-recommends php7.2-apcu php7.2-bcmath \
-        php7.2-cli php7.2-curl php7.2-fpm php7.2-gd php7.2-intl php7.2-mysql php7.2-xml \
-        php7.2-zip php7.2-mbstring nodejs yarn && \
+    apt-get install -y --no-install-recommends php7.3-apcu php7.3-bcmath php7.3-cli php7.3-curl php7.3-fpm php7.3-gd php7.3-intl php7.3-mysql php7.3-xml php7.3-zip php7.3-mbstring nodejs yarnpkg && \
```
Comment (pierallard) [inline Dockerfile]: Do you still need PHP?

File: Dockerfile [hunk 2]
```diff
@@ -68,19 +57,13 @@ RUN chmod +x /home/akeneo/pim-docs/build.sh && \
...
-RUN service mysql start && chmod +x /wait_for_mysql.sh && /wait_for_mysql.sh && \
-    mysql -u root -e "CREATE DATABASE akeneo_pim" && \
-    mysql -u root -e "GRANT ALL PRIVILEGES ON akeneo_pim.* TO akeneo_pim@localhost IDENTIFIED BY 'akeneo_pim'" && \
-    cp /home/akeneo/pim-docs/pim-community-dev-master/app/config/parameters.yml.dist /home/akeneo/pim-docs/pim-community-dev-master/app/config/parameters.yml && \
-    cd /home/akeneo/pim-docs/pim-community-dev-master/ && php bin/console doctrine:schema:create --env=prod && \
-    cd /home/akeneo/pim-docs/pim-community-dev-master/ && php bin/console pim:installer:assets --env=prod && \
+RUN cd /home/akeneo/pim-docs/pim-community-dev-master/ && php bin/console pim:installer:assets --env=prod && \
     cd /home/akeneo/pim-docs/pim-community-dev-master/ && sed -i "s#replace: '/bundles'#replace: '../bundles'#" frontend/build/compile-less.js && \
-    cd /home/akeneo/pim-docs/pim-community-dev-master/ && yarn install && yarn less
+    cd /home/akeneo/pim-docs/pim-community-dev-master/ && yarnpkg install && yarnpkg less
```
Comment (pierallard) [inline Dockerfile]: Do you still need yarn ?

File: README.md [hunk 3]
```diff
@@ -2,11 +2,38 @@
...
+
+From the `./pim-docs` directory, run:
+
+```bash
+    $ docker build . --tag pim-docs:master
+    $ rm -rf pim-docs-build && mkdir pim-docs-build
+    $ docker run --rm \
+        -v $(pwd):/home/akeneo/pim-docs/data \
+        pim-docs:master \
+        ./build.sh master --uid $(id -u) --gid $(id -g) --no-asset-check
```
Comment (pierallard) [inline README.md]: (You removed the no-asset-check)

File: README.md [hunk 4]
```diff
@@ -2,11 +2,38 @@
...
+## Build the documentation
+
+### With Docker
+
+Install [Docker](https://docs.docker.com/engine/installation/).
+
+[optional] To update the branch list with the current pim-docs branches, use:
+
+```
+sed -i -e "s/^\(.*\)'versions': .*,\(.*\)$/\1'versions': ['$(git branch -l|grep -x "\(^[ *]\+[0-9]\+\.[0-9]\+.*\)\|\(^[ *]\+master\)" | cut -c 3- | sort -r | paste -sd " ")'],\2/" conf.py
```
Comment (pierallard) [inline README.md]: This is not true anymore. You can drop this.
Versions are done with this
https://github.com/akeneo/pim-docs/pull/1208/files#diff-c09a020a80f3e0d8babc053a99898133

File: README.md (hunk 3 above)
Comment (jjanvier) [reply README.md]: yes, as now we always download the `pim.css`

File: Dockerfile (hunk 2 above)
Comment (jjanvier) [reply Dockerfile]: I still launch the `yarn less` command
not sure it's needed
I'll test tomorrow

File: Dockerfile (hunk 1 above)
Comment (jjanvier) [reply Dockerfile]: I still launch the `pim:install:assets` command
not sure it's needed
I'll test tomorrow

**Pull request merged**